src/lib.rs

```diff
@@ -553,47 +570,7 @@ fn advance_8byte_word(word: u64, bytes: usize) -> u64 {
 }
 
 #[inline]
-fn compare_masked(left: u64, right: u64, ignored_bits: u16) -> bool {
-    let mask = if ignored_bits == 64 {
-        0
-    } else {
-        u64::MAX >> ignored_bits
-    };
-
+pub(crate) fn compare_masked(left: u64, right: u64, ignored_bits: u16) -> bool {
+    let mask = u64::MAX >> ignored_bits;
     (left & mask) == right
 }
-
-/// This is a function that will get monomorphized based on the value of `N` to do
-/// a load of `N` values from the pointer in a minimum number of instructions into
-/// an output `u64`.
-#[inline]
-unsafe fn extract_u64<const N: usize>(ptr: *const u8) -> u64 {
```

Comment:
With better benchmarking it became clear this had no effect on performance VS just doing a copy_non_overlapping